### PR TITLE
Fix network dropdown button

### PR DESCRIPTION
**Pull request by Gudahtt**
This fixes a bug where the network menu would remain present after a second click on the network menu button. The bug was caused by the click being handled _twice_, by two separate handlers. First it was caught by the external click handler of the dropdown menu, which closed the menu. Second, it was caught by the network button itself, which re-opened the menu. This all happens quickly enough that to the user it appears to stay open.

The external click handler of the menu now only fires if the menu is open. Additionally, any click that is caught by the network menu is stopped from propagating further, so that it can't trigger additional click handlers.

Manual testing steps:  
  - Click the network menu button
  - Click it again. See that it closes the network menu.

### CODE REVIEW

**Comment by Gudahtt** [comment]
This depends upon #10311

**Review by brad-decker — DISMISSED** [review]
LGTM, i'll reapprove after the other one merges.

**Review by brad-decker — APPROVED** [review]

**Pull request merged**
